terms/hyperparameters_nm.tex

```diff
@@ -0,0 +1,70 @@
+
+
+\subsection{Hyperparameter Optimization Through Adaptive Resource Allocation}
+ As stated by Li \textit{et al.} in a more recently published paper\cite{Li2017}, the interaction between hyperparameters themselves is often not well understood. Generally, the main point of interest in a system is how the hyperparameters influence its performance, rather than how they affect each other as one configuration. This leads to brute force methods being used, such as the grid search method mentioned above, and random searching. Whilst these methods may be used to find good values of hyperparameters, they will more than likely result in suboptimal settings, especially for larger scale models. Bayesian techniques aim to optimize hyperparameter configurations by selecting them adaptively, beating out brute-force methods. Adaptive resource allocation is one such method which is used for selection. Strong hyperparameter configurations bearing positive results are provided more resources, such as an increased amount of training attributes. Other reward mechanisms include a larger number of features and epochs being provided. Other settings with less promising results may be discarded or allotted an inferior amount of resources. 
+
+\subsection{Bandit-Based Approach in Conjunction with Adaptive Allocation}
+ The HYPERBAND system was constructed by the same authors as above. It attempts to evaluate randomly-sampled hyperparameter configurations by adaptively assigning the corresponding amount of resources according to validation losses. This is termed a bandit-based approach since the project makes use of 'infinite armed bandits' in its implementation. Bandit arms represent mean-level reward distributions from particular experiments. The reward budget of each arm is essential for selection of the best distributions.  As detailed by Wang \textit{et al.} in \cite{Wang2009}, multi-armed bandit problems focus on \textit{exploration} vs. \textit{exploitation}. Arms can be `exploited', meaning that they performed well and may be selected for their respective task. Exploration may entail pulling entirely new arms, or pulling already discovered ones of interest. Scoping back to the HYPERBAND paper, the focus is on exploration infinite-armed bandit problems, selecting arms within the top reward percentiles for hyperparameter optimization.
+
+
+\section{Scenario: Hyperparameter Tuning for Gradient Descent}
+This scenario will look into a practical example of how hyperparameter values affect the performance of a gradient descent algorithm. Consider the following setup with cost function:
+
+\begin{eqnarray}
+y = 4x^{2} - 2x - 1 \\
+dy/{dx} = 8x-2
```

Comment:
use proper dy/dx - see [here](https://www.maths.tcd.ie/~dwilkins/LaTeXPrimer/Calculus.html).
Equations need not be numbered if not referenced from text (and you do not need to reference everything from text).